fix(Config Schema): validate resourcePolicy is array

glb opened the pull request:
In the AWS provider, the `resourcePolicy` must be an array. If you paste a resource policy from the console, your stack will fail to create.

Closes: #7795 
Addresses: #8018

glb commented:
I don't love that the test takes so long to run but I don't know a faster way to do it; based on tests in #7335. If there is a better way to test this I'd be happy to make the change.

medikoo commented on lib/plugins/aws/provider/awsProvider.test.js:
```diff
@@ -17,6 +17,8 @@ const Serverless = require('../../../Serverless');
 const { replaceEnv } = require('../../../../tests/utils/misc');
 const { getTmpFilePath } = require('../../../../tests/utils/fs');
 
+const runServerless = require('../../../../tests/utils/run-serverless');
```
Let's skip testing that. AJV is extensively tested on its own, we do not have to double confirm that `{ type: 'array' }` works, and sanity testing (that AJV is setup property) is already made in basic schema tests.

medikoo commented on lib/plugins/aws/provider/awsProvider.js:
```diff
@@ -140,6 +140,11 @@ class AwsProvider {
       serverless.configSchemaHandler.defineProvider('aws', {
+        provider: {
+          properties: {
+            resourcePolicy: { type: 'array' },
```
Maybe we can make it a bit more precise? If I see correctly each item should be an object, therefore maybe let's add `items: { type: 'object' }`

glb commented:
Thanks @medikoo ! I have updated, rebased, and done manual testing with invalid and valid values for `resourcePolicy`:

<details>
<summary>serverless.yml</summary>

```yaml
service: test

provider:
  name: aws
  resourcePolicy:
    - test
```

</details>

Output:

```sh
Serverless: Configuration error at 'provider.resourcePolicy[0]': should be object
```

<details>
<summary>serverless.yml</summary>

```yaml
service: test

provider:
  name: aws
  resourcePolicy: {}
```

</details>

Output:

```sh
Serverless: Configuration error at 'provider.resourcePolicy': should be array
```

<details>
<summary>serverless.yml</summary>

```yaml
service: test

provider:
  name: aws
  resourcePolicy: 
  - Effect: Allow
    Principal: "*"
    Action: "*"
    Resource: "*"
```

</details>

Output: _no error_ (though arguably that's a terrible resource policy. 😆 )
